src/webui/www/private/scripts/mocha-init.js

```diff
@@ -658,7 +667,150 @@ var initializeWindows = function() {
     };
 
     deleteTorrentsByCategoryFN = function(categoryHash) {
-        var hashes = torrentsTable.getFilteredTorrentsHashes('all', categoryHash);
+        var hashes = torrentsTable.getFilteredTorrentsHashes('all', categoryHash, TAGS_ALL);
+        if (hashes.length) {
+            new MochaUI.Window({
+                id: 'confirmDeletionPage',
+                title: "QBT_TR(Deletion confirmation)QBT_TR[CONTEXT=confirmDeletionDlg]",
+                loadMethod: 'iframe',
+                contentURL: 'confirmdeletion.html?hashes=' + hashes.join("|"),
+                scrollbars: false,
+                resizable: false,
+                maximizable: false,
+                padding: 10,
+                width: 424,
+                height: 140
+            });
+            updateMainData();
+        }
+    };
+
+    torrentAddTagsFN = function() {
+        const action = "set";
+        const hashes = torrentsTable.selectedRowsIds();
+        if (hashes.length) {
+            new MochaUI.Window({
+                id: 'newTagPage',
+                title: "QBT_TR(Add Tags)QBT_TR[CONTEXT=TransferListWidget]",
+                loadMethod: 'iframe',
+                contentURL: 'newtag.html?action=' + action + '&hashes=' + hashes.join('|'),
+                scrollbars: false,
+                resizable: false,
+                maximizable: false,
+                paddingVertical: 0,
+                paddingHorizontal: 0,
+                width: 250,
+                height: 100
+            });
+        }
+    };
+
+    torrentSetTagsFN = function(tagHash, isSet) {
+        let tagName = '';
```

Comment:
Nitpick, but can we make this `const` using a ternary operator?

`const tagName  = ((tagHash === 0) ? '' : tagList[tagHash].name);`

Note that I've switched from double equals to triple equals, but please confirm this is valid.